### DISCUSSION TITLE
Is it possible to process multi-yaml files with clj-yaml?

### DISCUSSION THREAD

**Discussion by underside** [clj-commons/clj-yaml#111] [General]
I have started to use babashka for my DevOps tasks and face an issue/misunderstanding.

I tried to parse multi-yaml(one file with few yaml manifests separated by ---) file using babashka and clj-yaml and got an error

- yaml  file something like 
```

# Logging config
logging: file

---
# Other config in same file
monitoring-tag: static
```

- How I tried to parse this
```
(let [conf-path (format "%s/configmap/application.yml" <path_to_file> ))
      conf (yaml/parse-string (slurp conf-path))]
    (println (-> conf :key)))
```

- error
```
Type:     org.yaml.snakeyaml.composer.ComposerException
Message:  expected a single document in the stream

```

Is it possible to process multi-yaml files with clj-yaml?  It's quite critical for me, since most of our configuration files is multi-yaml.

**Comment by borkdude** [comment]
@underside Without knowing too much about this, I did find a multi-doc test here:

https://github.com/clj-commons/clj-yaml/blob/e847e3e9ab9b847bb21806ad378273ccf209486b/test/clj_yaml/core_test.clj#L341

Hopefully it helps looking at this example. If not, feel free to come back with more questions and we'll figure something out.

**Reply by borkdude** [reply]
@underside The `:load-all` option seems to be necessary.

**Comment by underside** [comment]
Thanks @borkdude for help and babashka! 

I have just started to learn Clojure and babashka is awesome project for me to help with this.
